src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Directory.Move.cs

```diff
@@ -0,0 +1,220 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class Directory_Move_Tests : FileSystemWatcherTest
+    {
+        #region WindowsTests
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void Windows_Directory_Move_To_Same_Directory()
+        {
+            DirectoryMove_SameDirectory(WatcherChangeTypes.Renamed);
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void Windows_Directory_Move_From_Watched_To_Unwatched()
+        {
+            DirectoryMove_FromWatchedToUnwatched(WatcherChangeTypes.Deleted);
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void Windows_Directory_Move_To_Different_Watched_Directory()
+        {
+            DirectoryMove_DifferentWatchedDirectory(WatcherChangeTypes.Changed);
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void Windows_Directory_Move_From_Unwatched_To_Watched()
+        {
+            DirectoryMove_FromUnwatchedToWatched(WatcherChangeTypes.Created);
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(false)]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void Windows_Directory_Move_In_Nested_Directory(bool includeSubdirectories)
+        {
+            DirectoryMove_NestedDirectory(includeSubdirectories ? WatcherChangeTypes.Changed | WatcherChangeTypes.Renamed : 0, includeSubdirectories);
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void Windows_Directory_Move_With_Set_NotifyFilter()
+        {
+            DirectoryMove_WithNotifyFilter(WatcherChangeTypes.Renamed);
+        }
+
+        #endregion
+
+        #region UnixTests
+
+        [Fact]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        public void Unix_Directory_Move_To_Same_Directory()
+        {
+            DirectoryMove_SameDirectory(WatcherChangeTypes.Renamed);
```

Comment:
A few reasons:
- The OSX tests used to have different results than they do now and I wrote the tests first.
- laziness: I copy-pasted the Windows tests and added Unix to the names.
- They were already split like this in the original tests so I figured "Hey, might as well be consistent"

So there's no good reason why they should be separate tests.

Digression: it would be cool if we had Platform-specific InlineData e.g.
```
[Theory]
[InlineData(PlatformID.Linux, WatcherChangeTypes.Deleted | WatcherChangeTypes.Created)]
[InlineData(PlatformID.OSX, WatcherChangeTypes.Created)]
[InlineData(PlatformID.Windows, WatcherChangeTypes.Renamed)]
public void Directory_Move_To_Same_Directory(WatcherChangeTypes expected) {...}
```

so we wouldn't need different methods for small differences like this.